### DISCUSSION TITLE
Bitwarden app login error from ios with selfhosted

### DISCUSSION THREAD

**Discussion by mauroruffino** [dani-garcia/vaultwarden#4965] [Q&A] [answered]
Hi everybody,
i use a casaos vaultwarden docker (latest version) with letsencrypt ca (with nginx) and the app works fine from android with https://...duckdns... url.

But bitwarden app newly installed on iphone 14plus only logs to bitwarden.com and not to https://...duckdns.... selfhosted url

anyone can helps me ?
thanks
bye
mauro

**Comment by dciriya** [comment]
I recently experienced the same issue after upgrading the Bitwarden app on iOS to the latest version. The version of the vaultwarden I'm self-hosting was 1.2x, but recently I upgraded the server to the 1.32.0 version of docker image and successfully logged in from the app to my self-hosting server. Just in case, please try replacing the docker image on the self-hosting server with the latest 1.32.x image or higher. In my case, the problem was completely resolved after upgrading the version of the server.

**Reply by mcblum** [reply]
So here's a strange thing, I have updated to the latest version, 1.32.0, and I'm still seeing the error over and over. Do you happen to know what the root cause is? Every client is fine except for iOS.

**Reply by mauroruffino** [reply]
bad news @mcblum ... have U tried from safari ? its the same also from safari ? or only from bitwarden app ?

**Reply by BlackDex** [reply]
Try the `testing` tagged images maybe?
Or try to fully purge/remove the app including data. This also happens sometimes on Android for some reason. That the cached data causes issues because of previously working, but in a new version wrong sync data.

Deleting all info and getting fresh data from the latest (or testing) container images might solve this.

**Comment by kcris** [comment]
upgraded to latest docker vaultwarden and still having the same issue on ios 18. 

Login does not work.

**Reply by BlackDex** [reply]
As mentioned in an other discussion, use the `testing` tagged containers which have a fix specifically for iOS clients. A stable release will be created soon.

**Reply by kcris** [reply]
tried `vaultwarden/server:testing`. Still getting the same error on ios18. Thanks

**Reply by BlackDex** [reply]
Any logs maybe. Just shouting it doesn't work, doesn't help.

**Reply by kcris** [reply, marked as answer]
I am not shouting. I had already posted the logs in another [thread](https://github.com/dani-garcia/vaultwarden/discussions/5026#discussioncomment-10969989).

Updated again and got the `Version 1.32.2-65629a99` now and it works. Thank you

**Reply by mauroruffino** [reply]
good news @kcris , i update my selfhosted vaultwarden docker compose to 1.32.2..... and now good news on ios 17.7 it works !!!!! thank you

**Comment by sebastienclaude** [comment]
For anyone hitting this **login loop on the mobile app with a self-hosted server** — the app authenticates, briefly shows a screen, then bounces back to the login screen, while the web vault works fine:

In our case the root cause was simply an **outdated Vaultwarden**. The container was pinned to `vaultwarden/server:latest`, but the local image had never been re-pulled, so it was stuck on **1.34.3 (~12 months old)**. Every server response was `200 OK` — login, `prelogin`, `/api/connect/token`, and `/api/sync` all succeeded (the vault was even fetched) — yet the recent Bitwarden mobile app kept looping back to the login screen. Clock (NTP) and TLS were both fine.

`docker compose pull && docker compose up -d` to upgrade to **1.37.0** fixed it instantly. The SQLite DB auto-migrated without any issue (back up your data volume first, just in case).

Gotcha worth highlighting for others: **recreating a `:latest` container does NOT pull a newer image** — you have to `docker compose pull` explicitly. It's easy to silently run many months behind, which is exactly what triggered the incompatibility with the current mobile app here.
